### PR TITLE
LE Audio Split and update broadcast sink API

### PR DESCRIPTION
**Pull request by Thalley**
Updates the broadcast sink API to be completely separate from the connected audio API, and updates a few function names.

### CODE REVIEW

File: include/bluetooth/audio.h [hunk 1]
```diff
@@ -921,7 +921,7 @@ int bt_audio_chan_stop(struct bt_audio_chan *chan);
  *  source channel.
  *
  *  Broadcast sink channels shall be released using
```
Comment (asbjornsabo) [inline include/bluetooth/audio.h:923]: It sound weird to say that they shall be released using a delete function.
Propose to use same wording as for source (just below): "cannot be released, but can be deleted [...]"

File: include/bluetooth/audio.h [hunk 2]
```diff
@@ -1097,15 +1097,15 @@ int bt_audio_broadcast_sink_sync(struct bt_audio_broadcast_sink *sink,
 /** @brief Release a broadcast sink
  *
  *  Once a broadcast sink has been allocated after the pa_synced callback,
- *  it can be released using this function. If the sink has synchronized to any
+ *  it can be delete using this function. If the sink has synchronized to any
```
Comment (asbjornsabo) [inline include/bluetooth/audio.h:1112]: "deleted"

File: subsys/bluetooth/host/audio/chan.c [hunk 3]
```diff
@@ -734,7 +734,7 @@ int bt_audio_chan_release(struct bt_audio_chan *chan, bool cache)
 		BT_DBG("Cannot release a broadcast source");
 		return -EINVAL;
 	} else if (bt_audio_ep_is_broadcast_snk(chan->ep)) {
-		BT_DBG("Broadcast sink shall be released with bt_audio_broadcast_sink_release");
+		BT_DBG("Broadcast sink shall be released with bt_audio_broadcast_sink_delete");
```
Comment (asbjornsabo) [inline subsys/bluetooth/host/audio/chan.c:728]: Same release/delete.  Suggest to reuse wording for source here as well.

File: subsys/bluetooth/host/audio/chan.c [hunk 4]
```diff
@@ -2477,10 +2468,41 @@ int bt_audio_broadcast_sink_sync(struct bt_audio_broadcast_sink *sink,
...
+
+	CHECKIF(sink == NULL) {
+		BT_DBG("sink is NULL");
+		return -EINVAL;
+	}
+
+	chan = sink->chan;
+
+	if (chan->state != BT_AUDIO_CHAN_STREAMING ||
+	    chan->state != BT_AUDIO_CHAN_CONFIGURED) {
```
Comment (asbjornsabo) [inline subsys/bluetooth/host/audio/chan.c:2495]: Should be AND ("&&") - the state is going to be un-equal to at least one of them, so this test will always be true.

File: tests/bluetooth/bsim_bt/bsim_test_audio/src/broadcast_sink_test.c [hunk 5]
```diff
@@ -127,13 +157,53 @@ static void test_main(void)
...
+	printk("Broadcast source PA synced, waiting for BASE\n");
+	WAIT_FOR_FLAG(base_received);
+	printk("BASE received\n");
+
+	err = bt_audio_broadcast_sink_delete(g_sink);
+	if (err != 0) {
+		FAIL("Unable to delete sink: %d", err);
+		return;
+	}
+	/* No "sync lost" event is generated when we initialie the disconnect */
```
Comment (asbjornsabo) [inline tests/bluetooth/bsim_bt/bsim_test_audio/src/broadcast_sink_test.c:188]: Nitpicking: initialize - missing "z"

**Review by asbjornsabo — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/audio/chan.c (hunk 4 above)
Comment (Thalley) [reply subsys/bluetooth/host/audio/chan.c:2495]: Thanks!

File: include/bluetooth/audio.h (hunk 1 above)
Comment (asbjornsabo) [reply include/bluetooth/audio.h]: A missing "d" again ... 
(Nitpicking, I know.  But it looks better without spelling errors.)

**Review by asbjornsabo — APPROVED** [review]
One minor spelling correction left, otherwise LGTM

File: include/bluetooth/audio.h (hunk 1 above)
Comment (Thalley) [reply include/bluetooth/audio.h]: Odd that I keep making that typo... Oh well, it's fixed :D

**Pull request merged**